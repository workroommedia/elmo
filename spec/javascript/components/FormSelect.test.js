import React from "react";
import {shallow} from "enzyme";

import Component from "../../../app/javascript/components/FormSelect";

const changeFunc = jest.fn();

const defaultProps = {
  id: "root",
  options: [
    {id: "item1", name: "First"},
    {id: "item2", name: "Second"},
  ],
  changeFunc,
};

it("renders as expected", () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  expect(wrapper).toMatchSnapshot();
});

describe("after changing value", () => {
  const wrapper = shallow(<Component {...defaultProps} />);
  wrapper.find("select").simulate("change", {target: {value: defaultProps.options[0].id}});

  it("calls changeFunc", () => {
    expect(changeFunc).toMatchSnapshot();
  });

  it("renders as expected", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
