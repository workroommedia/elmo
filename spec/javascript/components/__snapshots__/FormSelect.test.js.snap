// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`after changing value calls changeFunc 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "item1",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": undefined,
    },
  ],
}
`;

exports[`after changing value renders as expected 1`] = `
<select
  className="form-control"
  defaultValue={null}
  id="root"
  key="root"
  name={null}
  onChange={[Function]}
>
  <option
    key="blank"
    value=""
  />
  <option
    key="item1"
    value="item1"
  >
    First
  </option>
  <option
    key="item2"
    value="item2"
  >
    Second
  </option>
</select>
`;

exports[`renders as expected 1`] = `
<select
  className="form-control"
  defaultValue={null}
  id="root"
  key="root"
  name={null}
  onChange={[Function]}
>
  <option
    key="blank"
    value=""
  />
  <option
    key="item1"
    value="item1"
  >
    First
  </option>
  <option
    key="item2"
    value="item2"
  >
    Second
  </option>
</select>
`;
